smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -205,10 +226,21 @@ public KafkaSource(Vertx vertx,
                         return this.consumer.subscribe(topics);
                     }
                 })
-                .map(rec -> commitHandler
-                        .received(
-                                new IncomingKafkaRecord<>(rec, commitHandler, failureHandler, isCloudEventEnabled,
-                                        isTracingEnabled)));
+                .onRequest().call(l -> {
+                    lastRequest.set(System.currentTimeMillis());
+                    if (paused.compareAndSet(true, false)) {
+                        log.resumingChannel(channel);
+                        return consumer.paused()
+                                .flatMap(consumer::resume);
```

Comment:
Not sure if this is related to the issues you saw with this PR.

Back when we first tried to allow for a user provided consumer re-balance listener we used **pause** and **resume** to get around the Vert.x client's non blocking interface. Using **resume** broke the consumer because it switched it from fetch mode to flow mode: 
> io.smallrye.mutiny.subscription.BackPressureFailure: Buffer full, cannot emit item

See Streaming mode:
https://vertx.io/docs/apidocs/io/vertx/core/streams/ReadStream.html

references:
https://github.com/vert-x3/vertx-kafka-client/pull/180
https://github.com/smallrye/smallrye-reactive-messaging/pull/672

Reply:
No, it's way worse than this. When paused, it stops polling, to after the heartbeat period, so the broker considers you "dead". 

In addition, the pause/resume methods from the client are unrelated to the pause/resume methods from Kafka. 

I have made progress on a new client in this PR: #995